src/codegen.cpp

```diff
@@ -2495,7 +2504,7 @@ static bool emit_builtin_call(jl_cgval_t *ret, jl_value_t *f, jl_value_t **args,
             if (jl_is_leaf_type(tp0)) {
                 emit_expr(args[1], ctx);
                 assert(jl_is_datatype(tp0));
-                *ret = mark_julia_type(ConstantInt::get(T_size, jl_datatype_nfields(tp0)), false, jl_long_type, ctx);
+                *ret = mark_julia_const(jl_box_long(jl_datatype_nfields(tp0)));
```

Comment:
This is the first one. The `jl_box_bool`s are fine since they never allocates. The `jl_box_long` with a small constant are also fine since they should always hit the cache too.

I'm not sure how these are used but basically you need to make sure that this is rooted as long as the value might be used by anyone. For value you need to embed in the generated code you need to push it to the roots array (there's a function for this) but I'm not sure what's the convention on `mark_julia_const` now.